src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -7,17 +7,49 @@
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.stream.Collectors;
+
 @Service
 @Transactional
 public class LineService {
-    private LineRepository lineRepository;
+    private final LineRepository lineRepository;
 
     public LineService(LineRepository lineRepository) {
         this.lineRepository = lineRepository;
     }
 
-    public LineResponse saveLine(LineRequest request) {
+    public LineResponse save(LineRequest request) {
         Line persistLine = lineRepository.save(request.toLine());
         return LineResponse.of(persistLine);
     }
+
+    public List<LineResponse> findAll() {
+        List<Line> lines = lineRepository.findAll();
+
+        return lines.stream()
+                .map(line -> LineResponse.of(line))
+                .collect(Collectors.toList());
+    }
+
+    public LineResponse findById(long id) {
+        Line line = find(id);
+        return LineResponse.of(line);
+    }
+
+    public void update(Long id, LineRequest lineRequest) {
+        Line line = find(id);
+        line.update(lineRequest.toLine());
+    }
+
+    private Line find(long id) {
+        return lineRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException());
```

Comment:
```suggestion
                .orElseThrow(EntityNotFoundException::new);
```
이렇게 하던지 아니면 적절한 메시지를 추가해줘도 좋을 것 같네요 :)

Reply:
훨씬 깔끔하네요 반영하도록 하겠습니다